### ISSUE TITLE
`Error: not a tracked temp dir` when deploying

### ISSUE THREAD

**Issue by fourseven** [meteor/meteor#3761]
I've had a few people raise issue with `fourseven:scss` (fourseven/meteor-scss#66 fourseven/meteor-scss#67) and it's not only with that package (https://github.com/meteor-velocity/html-reporter/issues/60). The stack dump is as follows:

```
/Users/Christian/.meteor/packages/meteor-tool/.1.0.40.959pzb++os.osx.x86_64+web.browser+web.cordova/meteor-tool-os.osx.x86_64/dev_bundle/lib/node_modules/fibers/future.js:173
                        throw(ex);
                              ^
Error: not a tracked temp dir: /Users/Christian/.meteor/packages/fourseven_scss/.0.9.6.hcwe4p++os.osx.x86_64+web.browser+web.cordova
    at Error (<anonymous>)
    at Object.files.freeTempDir (/Users/Christian/.meteor/packages/meteor-tool/.1.0.40.959pzb++os.osx.x86_64+web.browser+web.cordova/meteor-tool-os.osx.x86_64/tools/files.js:544:11)
    at /Users/Christian/.meteor/packages/meteor-tool/.1.0.40.959pzb++os.osx.x86_64+web.browser+web.cordova/meteor-tool-os.osx.x86_64/tools/tropohouse.js:348:17
    at Array.forEach (native)
    at Function._.each._.forEach (/Users/Christian/.meteor/packages/meteor-tool/.1.0.40.959pzb++os.osx.x86_64+web.browser+web.cordova/meteor-tool-os.osx.x86_64/dev_bundle/lib/node_modules/underscore/underscore.js:79:11)
    at /Users/Christian/.meteor/packages/meteor-tool/.1.0.40.959pzb++os.osx.x86_64+web.browser+web.cordova/meteor-tool-os.osx.x86_64/tools/tropohouse.js:347:11
    at /Users/Christian/.meteor/packages/meteor-tool/.1.0.40.959pzb++os.osx.x86_64+web.browser+web.cordova/meteor-tool-os.osx.x86_64/tools/buildmessage.js:352:18
    at [object Object]._.extend.withValue (/Users/Christian/.meteor/packages/meteor-tool/.1.0.40.959pzb++os.osx.x86_64+web.browser+web.cordova/meteor-tool-os.osx.x86_64/tools/fiber-helpers.js:115:14)
    at /Users/Christian/.meteor/packages/meteor-tool/.1.0.40.959pzb++os.osx.x86_64+web.browser+web.cordova/meteor-tool-os.osx.x86_64/tools/buildmessage.js:345:34
    at [object Object]._.extend.withValue (/Users/Christian/.meteor/packages/meteor-tool/.1.0.40.959pzb++os.osx.x86_64+web.browser+web.cordova/meteor-tool-os.osx.x86_64/tools/fiber-helpers.js:115:14)
    at /Users/Christian/.meteor/packages/meteor-tool/.1.0.40.959pzb++os.osx.x86_64+web.browser+web.cordova/meteor-tool-os.osx.x86_64/tools/buildmessage.js:343:23
    at [object Object]._.extend.withValue (/Users/Christian/.meteor/packages/meteor-tool/.1.0.40.959pzb++os.osx.x86_64+web.browser+web.cordova/meteor-tool-os.osx.x86_64/tools/fiber-helpers.js:115:14)
    at Object.enterJob (/Users/Christian/.meteor/packages/meteor-tool/.1.0.40.959pzb++os.osx.x86_64+web.browser+web.cordova/meteor-tool-os.osx.x86_64/tools/buildmessage.js:317:26)
    at Object.download (/Users/Christian/.meteor/packages/meteor-tool/.1.0.40.959pzb++os.osx.x86_64+web.browser+web.cordova/meteor-tool-os.osx.x86_64/tools/tropohouse.js:303:20)
    at /Users/Christian/.meteor/packages/meteor-tool/.1.0.40.959pzb++os.osx.x86_64+web.browser+web.cordova/meteor-tool-os.osx.x86_64/tools/tropohouse.js:412:22
    at /Users/Christian/.meteor/packages/meteor-tool/.1.0.40.959pzb++os.osx.x86_64+web.browser+web.cordova/meteor-tool-os.osx.x86_64/tools/buildmessage.js:352:18
    at [object Object]._.extend.withValue (/Users/Christian/.meteor/packages/meteor-tool/.1.0.40.959pzb++os.osx.x86_64+web.browser+web.cordova/meteor-tool-os.osx.x86_64/tools/fiber-helpers.js:115:14)
    at /Users/Christian/.meteor/packages/meteor-tool/.1.0.40.959pzb++os.osx.x86_64+web.browser+web.cordova/meteor-tool-os.osx.x86_64/tools/buildmessage.js:345:34
    at [object Object]._.extend.withValue (/Users/Christian/.meteor/packages/meteor-tool/.1.0.40.959pzb++os.osx.x86_64+web.browser+web.cordova/meteor-tool-os.osx.x86_64/tools/fiber-helpers.js:115:14)
    at /Users/Christian/.meteor/packages/meteor-tool/.1.0.40.959pzb++os.osx.x86_64+web.browser+web.cordova/meteor-tool-os.osx.x86_64/tools/buildmessage.js:343:23
    at [object Object]._.extend.withValue (/Users/Christian/.meteor/packages/meteor-tool/.1.0.40.959pzb++os.osx.x86_64+web.browser+web.cordova/meteor-tool-os.osx.x86_64/tools/fiber-helpers.js:115:14)
    at Object.enterJob (/Users/Christian/.meteor/packages/meteor-tool/.1.0.40.959pzb++os.osx.x86_64+web.browser+web.cordova/meteor-tool-os.osx.x86_64/tools/buildmessage.js:317:26)
    at [object Object]._.extend.downloadPackagesMissingFromMap (/Users/Christian/.meteor/packages/meteor-tool/.1.0.40.959pzb++os.osx.x86_64+web.browser+web.cordova/meteor-tool-os.osx.x86_64/tools/tropohouse.js:409:20)
    at /Users/Christian/.meteor/packages/meteor-tool/.1.0.40.959pzb++os.osx.x86_64+web.browser+web.cordova/meteor-tool-os.osx.x86_64/tools/project-context.js:600:25
    at /Users/Christian/.meteor/packages/meteor-tool/.1.0.40.959pzb++os.osx.x86_64+web.browser+web.cordova/meteor-tool-os.osx.x86_64/tools/buildmessage.js:352:18
    at [object Object]._.extend.withValue (/Users/Christian/.meteor/packages/meteor-tool/.1.0.40.959pzb++os.osx.x86_64+web.browser+web.cordova/meteor-tool-os.osx.x86_64/tools/fiber-helpers.js:115:14)
    at /Users/Christian/.meteor/packages/meteor-tool/.1.0.40.959pzb++os.osx.x86_64+web.browser+web.cordova/meteor-tool-os.osx.x86_64/tools/buildmessage.js:345:34
    at [object Object]._.extend.withValue (/Users/Christian/.meteor/packages/meteor-tool/.1.0.40.959pzb++os.osx.x86_64+web.browser+web.cordova/meteor-tool-os.osx.x86_64/tools/fiber-helpers.js:115:14)
    at /Users/Christian/.meteor/packages/meteor-tool/.1.0.40.959pzb++os.osx.x86_64+web.browser+web.cordova/meteor-tool-os.osx.x86_64/tools/buildmessage.js:343:23
    at [object Object]._.extend.withValue (/Users/Christian/.meteor/packages/meteor-tool/.1.0.40.959pzb++os.osx.x86_64+web.browser+web.cordova/meteor-tool-os.osx.x86_64/tools/fiber-helpers.js:115:14)
    at Object.enterJob (/Users/Christian/.meteor/packages/meteor-tool/.1.0.40.959pzb++os.osx.x86_64+web.browser+web.cordova/meteor-tool-os.osx.x86_64/tools/buildmessage.js:317:26)
    at /Users/Christian/.meteor/packages/meteor-tool/.1.0.40.959pzb++os.osx.x86_64+web.browser+web.cordova/meteor-tool-os.osx.x86_64/tools/project-context.js:599:20
    at /Users/Christian/.meteor/packages/meteor-tool/.1.0.40.959pzb++os.osx.x86_64+web.browser+web.cordova/meteor-tool-os.osx.x86_64/tools/buildmessage.js:264:13
    at [object Object]._.extend.withValue (/Users/Christian/.meteor/packages/meteor-tool/.1.0.40.959pzb++os.osx.x86_64+web.browser+web.cordova/meteor-tool-os.osx.x86_64/tools/fiber-helpers.js:115:14)
    at /Users/Christian/.meteor/packages/meteor-tool/.1.0.40.959pzb++os.osx.x86_64+web.browser+web.cordova/meteor-tool-os.osx.x86_64/tools/buildmessage.js:257:29
    at [object Object]._.extend.withValue (/Users/Christian/.meteor/packages/meteor-tool/.1.0.40.959pzb++os.osx.x86_64+web.browser+web.cordova/meteor-tool-os.osx.x86_64/tools/fiber-helpers.js:115:14)
    at /Users/Christian/.meteor/packages/meteor-tool/.1.0.40.959pzb++os.osx.x86_64+web.browser+web.cordova/meteor-tool-os.osx.x86_64/tools/buildmessage.js:255:18
    at [object Object]._.extend.withValue (/Users/Christian/.meteor/packages/meteor-tool/.1.0.40.959pzb++os.osx.x86_64+web.browser+web.cordova/meteor-tool-os.osx.x86_64/tools/fiber-helpers.js:115:14)
    at /Users/Christian/.meteor/packages/meteor-tool/.1.0.40.959pzb++os.osx.x86_64+web.browser+web.cordova/meteor-tool-os.osx.x86_64/tools/buildmessage.js:246:23
    at [object Object]._.extend.withValue (/Users/Christian/.meteor/packages/meteor-tool/.1.0.40.959pzb++os.osx.x86_64+web.browser+web.cordova/meteor-tool-os.osx.x86_64/tools/fiber-helpers.js:115:14)
    at Object.capture (/Users/Christian/.meteor/packages/meteor-tool/.1.0.40.959pzb++os.osx.x86_64+web.browser+web.cordova/meteor-tool-os.osx.x86_64/tools/buildmessage.js:245:19)
    at Object.catalog.runAndRetryWithRefreshIfHelpful (/Users/Christian/.meteor/packages/meteor-tool/.1.0.40.959pzb++os.osx.x86_64+web.browser+web.cordova/meteor-tool-os.osx.x86_64/tools/catalog.js:92:31)
    at [object Object]._.extend._downloadMissingPackages (/Users/Christian/.meteor/packages/meteor-tool/.1.0.40.959pzb++os.osx.x86_64+web.browser+web.cordova/meteor-tool-os.osx.x86_64/tools/project-context.js:598:13)
    at /Users/Christian/.meteor/packages/meteor-tool/.1.0.40.959pzb++os.osx.x86_64+web.browser+web.cordova/meteor-tool-os.osx.x86_64/tools/project-context.js:210:35
    at /Users/Christian/.meteor/packages/meteor-tool/.1.0.40.959pzb++os.osx.x86_64+web.browser+web.cordova/meteor-tool-os.osx.x86_64/tools/buildmessage.js:352:18
    at [object Object]._.extend.withValue (/Users/Christian/.meteor/packages/meteor-tool/.1.0.40.959pzb++os.osx.x86_64+web.browser+web.cordova/meteor-tool-os.osx.x86_64/tools/fiber-helpers.js:115:14)
    at /Users/Christian/.meteor/packages/meteor-tool/.1.0.40.959pzb++os.osx.x86_64+web.browser+web.cordova/meteor-tool-os.osx.x86_64/tools/buildmessage.js:345:34
    at [object Object]._.extend.withValue (/Users/Christian/.meteor/packages/meteor-tool/.1.0.40.959pzb++os.osx.x86_64+web.browser+web.cordova/meteor-tool-os.osx.x86_64/tools/fiber-helpers.js:115:14)
    at /Users/Christian/.meteor/packages/meteor-tool/.1.0.40.959pzb++os.osx.x86_64+web.browser+web.cordova/meteor-tool-os.osx.x86_64/tools/buildmessage.js:343:23
    at [object Object]._.extend.withValue (/Users/Christian/.meteor/packages/meteor-tool/.1.0.40.959pzb++os.osx.x86_64+web.browser+web.cordova/meteor-tool-os.osx.x86_64/tools/fiber-helpers.js:115:14)
    at Object.enterJob (/Users/Christian/.meteor/packages/meteor-tool/.1.0.40.959pzb++os.osx.x86_64+web.browser+web.cordova/meteor-tool-os.osx.x86_64/tools/buildmessage.js:317:26)
    at [object Object]._.extend._completeStagesThrough (/Users/Christian/.meteor/packages/meteor-tool/.1.0.40.959pzb++os.osx.x86_64+web.browser+web.cordova/meteor-tool-os.osx.x86_64/tools/project-context.js:200:18)
    at [object Object]._.extend.saveChangedMetadata (/Users/Christian/.meteor/packages/meteor-tool/.1.0.40.959pzb++os.osx.x86_64+web.browser+web.cordova/meteor-tool-os.osx.x86_64/tools/project-context.js:188:10)
    at [object Object]._.extend.prepareProjectForBuild (/Users/Christian/.meteor/packages/meteor-tool/.1.0.40.959pzb++os.osx.x86_64+web.browser+web.cordova/meteor-tool-os.osx.x86_64/tools/project-context.js:193:10)
    at /Users/Christian/.meteor/packages/meteor-tool/.1.0.40.959pzb++os.osx.x86_64+web.browser+web.cordova/meteor-tool-os.osx.x86_64/tools/commands.js:1078:20
    at /Users/Christian/.meteor/packages/meteor-tool/.1.0.40.959pzb++os.osx.x86_64+web.browser+web.cordova/meteor-tool-os.osx.x86_64/tools/buildmessage.js:264:13
    at [object Object]._.extend.withValue (/Users/Christian/.meteor/packages/meteor-tool/.1.0.40.959pzb++os.osx.x86_64+web.browser+web.cordova/meteor-tool-os.osx.x86_64/tools/fiber-helpers.js:115:14)
    at /Users/Christian/.meteor/packages/meteor-tool/.1.0.40.959pzb++os.osx.x86_64+web.browser+web.cordova/meteor-tool-os.osx.x86_64/tools/buildmessage.js:257:29
    at [object Object]._.extend.withValue (/Users/Christian/.meteor/packages/meteor-tool/.1.0.40.959pzb++os.osx.x86_64+web.browser+web.cordova/meteor-tool-os.osx.x86_64/tools/fiber-helpers.js:115:14)
    at /Users/Christian/.meteor/packages/meteor-tool/.1.0.40.959pzb++os.osx.x86_64+web.browser+web.cordova/meteor-tool-os.osx.x86_64/tools/buildmessage.js:255:18
    at [object Object]._.extend.withValue (/Users/Christian/.meteor/packages/meteor-tool/.1.0.40.959pzb++os.osx.x86_64+web.browser+web.cordova/meteor-tool-os.osx.x86_64/tools/fiber-helpers.js:115:14)
    at /Users/Christian/.meteor/packages/meteor-tool/.1.0.40.959pzb++os.osx.x86_64+web.browser+web.cordova/meteor-tool-os.osx.x86_64/tools/buildmessage.js:246:23
    at [object Object]._.extend.withValue (/Users/Christian/.meteor/packages/meteor-tool/.1.0.40.959pzb++os.osx.x86_64+web.browser+web.cordova/meteor-tool-os.osx.x86_64/tools/fiber-helpers.js:115:14)
    at Object.capture (/Users/Christian/.meteor/packages/meteor-tool/.1.0.40.959pzb++os.osx.x86_64+web.browser+web.cordova/meteor-tool-os.osx.x86_64/tools/buildmessage.js:245:19)
    at Object.main.captureAndExit (/Users/Christian/.meteor/packages/meteor-tool/.1.0.40.959pzb++os.osx.x86_64+web.browser+web.cordova/meteor-tool-os.osx.x86_64/tools/main.js:259:29)
    at Command.main.registerCommand.name [as func] (/Users/Christian/.meteor/packages/meteor-tool/.1.0.40.959pzb++os.osx.x86_64+web.browser+web.cordova/meteor-tool-os.osx.x86_64/tools/commands.js:1077:8)
    at /Users/Christian/.meteor/packages/meteor-tool/.1.0.40.959pzb++os.osx.x86_64+web.browser+web.cordova/meteor-tool-os.osx.x86_64/tools/main.js:1300:23
```

**Comment by mmw109Git** [comment]
I've got this same issue with collectionFS

Error: not a tracked temp dir: /Users/<myname>/.meteor/packages/cfs_gridfs/.0.0.27.1941v96++os.osx.x86_64+web.browser+web.cordova
    at Error (<anonymous>)
...

**Comment by hwillson** [comment]
Hi guys - I just encountered this issue as well. Looking at the code, I tracked the issue down here:

https://github.com/meteor/meteor/blob/38e0e344c439f991eb5153c6ec694e5e1dd12e82/tools/tropohouse.js (lines 307-312):

```
// If there's already a package in the tropohouse, start with it.      
if (packageLinkTarget) {                                               
  buildTempDirs.push(                                                  
    files.pathResolve(files.pathDirname(packageLinkFile),              
                      packageLinkTarget));                             
} 
```

The package path is getting added to buildTempDirs here; there is then an attempt to delete this directory shortly after (lines 346-349):

```
// Delete temp directories now (asynchronously).                       
_.each(buildTempDirs, function (buildTempDir) {                        
  files.freeTempDir(buildTempDir);                                     
}); 
```

The problem is that the path wasn't added using files.mkdtemp, which means it wasn't being tracked in the internal tempDirs array, and therefore can't be removed using files.freeTempDir (which then throws the "not a tracked temp dir" error).

I have some good news though - it looks like this issue has already been fixed in the devel branch - see 
https://github.com/meteor/meteor/commit/a03dc12be71d5cbeb0da6a9ee8146881193efefc

Hope this helps - thanks!

**Comment by bitsofinfo** [comment]
Same issue here

**Comment by christiansakai** [comment]
How do I use the development branch for my meteor app?

**Comment by dan-mesa** [comment]
@hwillson sounds like you tracked it down, but FWIW I did a `meteor logout` followed by a `meteor login` then deployed without issue. The good 'ol reboot.

**Comment by glasser** [comment]
Yeah, this will be out in 1.0.4.  My impression is that this bug is essentially self-healing --- it's a crash *after* downloading the second architecture of a package, but it downloads it successfully.  If you're hitting this problem with deploy, you will probably also hit with with `meteor build --architecture os.linux.x86_64 foo.tgz`, which should be faster because it doesn't upload anything. So as a workaround, run that command a few times until it works, and then you should be good until you add another binary package.

**Comment by glasser** [comment]
(closing since fixed on devel)

**Comment by glasser** [comment]
(If anyone *doesn't* see the workaround above as working, let me know!)

**Comment by christiansakai** [comment]
Hi glasser, I tried your approach, and the app keeps giving me error everytime

**Comment by glasser** [comment]
Is it the same error, with the same package name being mentioned each time (eg scss?)

**Comment by glasser** [comment]
(BTW, another workaround is to run `meteor build --architecture os.linux.x86_64 --release 1.0.2.1 x.tgz` once: this will use the 1.0.2.1 package downloader to download the Linux version of the package, which doesn't have the downloader bug. You can then continue to use 1.0.3.1 to deploy once the package is downloaded.)

**Comment by olizilla** [comment]
This appears to be affecting deploys to modulus for people using the cfs:gridfs package

**Comment by tgoorden** [comment]
This seems to work too, without too much hassle. You have to manually downgrade the meteor version to 1.0.2.1, by editing `.meteor/release` and changing the version number. (Until the permanent fix is released, don't update...)

**Comment by glasser** [comment]
I'm going to put out a 1.0.3.2 fixing only this.  See #3809.

**Comment by glasser** [comment]
(It would be great if people seeing this issue could test that the RC on #3809 fixes the issue!  It does for me.)

**Comment by glasser** [comment]
Should be fixed in 1.0.3.2, released today.

**Comment by realdlee** [comment]
I have version 1.0.3.2 and still see this error.
